eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/internal/IterableIterate.java

```diff
@@ -1100,7 +1108,7 @@ public static boolean notEmpty(Iterable<?> iterable)
         }
     }
 
-    public static <T> void forEachWithIndex(List<T> iterable, int from, int to, ObjectIntProcedure<? super T> objectIntProcedure)
+    public static <T> void forEachWithIndex(Iterable<T> iterable, int from, int to, ObjectIntProcedure<? super T> objectIntProcedure)
```

Comment:
Oof. This is a little thing to push us to a major-version, but it's technically a breaking change. I just want @donraab and @nikhilnanivadekar to see this one.

Reply:
Can't we just add a new static method which takes `Iterable`?

Reply:
it is called only once in the newly introduced `ImmutableSortedSetFromBagAdapter` which has an object of type `RichIterable` only. But note that we are in a class inside an `internal` package, shouldn't that be alright? Also I thought that it would be non-breaking because `Iterable` is a supertype of `List`, but maybe I'm missing something about Java type compatibility in bytecode...

Reply:
I would be conservative and make this kind of change in a major version. In the near term I would inline the code the code in the one place it is used or add another method on IterableIterate. @motlin what do you think?

Reply:
HI @victornoel I'd like to get this unblocked, as you and @motlin were making good progress through this review. Let's just add the code in the one place it is needed for now and we can update `IterableIterate` to take `Iterable` when we do a major release. Ok?

Reply:
@donraab sorry for the delay, so I was thinking: what about simply duplicating the method with `Iterable` as a parameter, I think that the overloading will only be taken into account at compile time, so it shouldn't have any impact on current uses of the old method. We can also deprecate the latest so that in the next major version with remove it?

Reply:
@motlin @donraab I went with this solution for now, but I can change it for something more ad-hoc if it's not correct :)